compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -730,21 +740,44 @@ convertBind env (name, x)
     | ConstraintTupleProjectionName _ _ <- name
     = pure []
 
+    -- Typeclass instance dictionaries
+    | DFunId isNewtype <- idDetails name
+    = withRange (convNameLoc name) $ do
+    x' <- convertExpr env x
+    -- NOTE(MH): This is DICTIONARY SANITIZATION step (3).
+    -- The sanitization for newtype dictionaries is done in `convertCoercion`.
+    let sanitized_x'
+          | isNewtype = x'
+          | otherwise =
+            let fieldsPrism
+                  | envLfVersion env `supports` featureTypeSynonyms = _EStructCon
+                  | otherwise = _ERecCon . _2
+            in over (_ETyLams . _2 . _ETmLams . _2 . fieldsPrism . each . _2) (ETmLam (mkVar "_", TUnit)) x'
+    name' <- convValWithType env name
+
+    -- OVERLAP* annotations
+    let overlapModeName = ExprValName ("$$om" <> getOccText name)
+        overlapModeValueM =
+            case MS.lookup name (envModInstanceInfo env) of
+                Nothing -> Nothing
+                Just (NoOverlap _) -> Nothing
+                Just (Overlappable _) -> Just "OVERLAPPABLE"
+                Just (Overlapping _) -> Just "OVERLAPPING"
+                Just (Overlaps _) -> Just "OVERLAPS"
+                Just (Incoherent _) -> Just "INCOHERENT"
```

Comment:
Do we really want to support this?

Reply:
As far as damlc is concerned, it's an "overlap mode" pragma like any other. In this light, I don't have any problem with it, or any reason to oppose it, compared to any other overlap modes.

One okay reason to support it is because instances marked "INCOHERENT" always have the lowest priority, so they have a kind of a unique function as an "ultimate default option", if you're going to be relying on overlapping instances at all.

Reply:
But I don't feel strongly either way. It is currently less work to support it than to emit a warning.